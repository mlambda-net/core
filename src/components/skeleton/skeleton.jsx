import React from 'react';
import PropTypes from 'prop-types';
import LocalizedStrings from 'react-localization';
import { Box } from '@mui/material';
import withUtils from '@mlambda-net/core/utils/withUtils';

const language = new LocalizedStrings({
  en: {},
  es: {},
});

const styles = (theme) => {
  return {};
};

class Skeleton extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    language.setLanguage(this.props.lang);
    return <Box width="100%">Skeleton</Box>;
  }
}

Skeleton.protoTypes = {
  className: PropTypes.string,
  classes: PropTypes.object.isRequired,
  lang: PropTypes.string,
};

export default withUtils(styles)(Skeleton);
