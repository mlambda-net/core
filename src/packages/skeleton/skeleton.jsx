import React from 'react';
import { withStyles, withTheme } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import LocalizedStrings from 'react-localization';
import withLanguage from '@mlambda-net/core/lang/language';
import { Valid } from '@mlambda-net/core/common/validations';
import Box from '@material-ui/core/Box';

const valid = Valid;

const language = new LocalizedStrings({
  en: {
  },
  es: {
  },
});

const styles = (theme) => {
  console.log(theme)
  return ({})
};

class Skeleton extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }


  render() {
    language.setLanguage(this.props.lang);
    return (
      <Box width="100%">
        Skeleton
      </Box>
    );
  }
}

Skeleton.protoTypes = {
  className: PropTypes.string,
  classes: PropTypes.object.isRequired,
  lang: PropTypes.string,
};

export default  withStyles(styles)(withTheme(withLanguage(Skeleton)));
