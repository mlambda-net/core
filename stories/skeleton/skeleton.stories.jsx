import React from 'react';
import { Skeleton } from '@mlambda-net/web-core/skeleton';

export default {
  title: 'MLambda/template/skeleton',
  component: Skeleton,
  tags: ['autodocs'],
};

export const Primary = {
  args: {
    name: 'welcome',
  },
  render: (args) => <Skeleton {...args} />,
};
