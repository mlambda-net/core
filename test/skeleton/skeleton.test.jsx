import React from 'react';
import renderer from 'react-test-renderer';
import { expect, test } from "@jest/globals";
import { Skeleton } from '@mlambda-net/core/skeleton';
import Wrap from "../util";

test('Skeleton render', () => {
  const item = renderer.create(
    <Wrap>
      <Skeleton />
    </Wrap>
  );
  let tree = item.toJSON();
  expect(tree).toMatchSnapshot();
});
