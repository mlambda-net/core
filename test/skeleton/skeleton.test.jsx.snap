// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton render 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-8atqhb{width:100%;}",
      }
    }
    data-emotion="css 8atqhb"
  />,
  <div
    className="MuiBox-root css-8atqhb"
  >
    Skeleton
  </div>,
]
`;
